clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -316,20 +325,29 @@ private MetadataCache handleMetadataResponse(MetadataResponse metadataResponse,
         Set<String> invalidTopics = new HashSet<>();
 
         List<MetadataResponse.PartitionMetadata> partitions = new ArrayList<>();
+        Map<String, Uuid> topicIds = new HashMap<>();
         for (MetadataResponse.TopicMetadata metadata : metadataResponse.topicMetadata()) {
-            topics.add(metadata.topic());
+            String topicName = metadata.topic();
+            Uuid topicId = metadata.topicId();
+            topics.add(topicName);
+            // We only update if the current metadata since we can only compare when both topics have valid IDs
+            Uuid oldTopicId = null;
+            if (!topicId.equals(Uuid.ZERO_UUID)) {
```

Comment:
nit: I think we're guaranteed that `topicId` is not null (in spite of the inconsistent `equals`), but it's still a little nicer to write this check as `!Uuid.ZERO_UUID.equals(topicId)`

Reply:
Yeah. There can not be a null uuid in the request. but I can rephrase